services/dynamic-sidecar/tests/unit/test_core_settings.py

```diff
@@ -1,13 +1,67 @@
 # pylint: disable=unused-argument
 # pylint: disable=redefined-outer-name
 
-
+import pytest
+from pytest_simcore.helpers.utils_envs import EnvVarsDict, setenvs_from_dict
 from simcore_service_dynamic_sidecar.core.settings import ApplicationSettings
 
 
-def test_settings_with_mock_environment(mock_environment):
+def test_settings_with_mock_environment(mock_environment: EnvVarsDict):
     assert ApplicationSettings.create_from_envs()
 
 
-def test_settings_with_envdevel_file(mock_environment_with_envdevel):
+def test_settings_with_envdevel_file(mock_environment_with_envdevel: EnvVarsDict):
     assert ApplicationSettings.create_from_envs()
+
+
+@pytest.mark.parametrize(
+    "envs",
+    [
+        {
+            "NODE_PORTS_STORAGE_LOGIN": "login",
+            "NODE_PORTS_STORAGE_PASSWORD": "passwd",
+            "NODE_PORTS_STORAGE_HOST": "host",
+            "NODE_PORTS_STORAGE_PORT": "42",
+        },
+        {
+            "NODE_PORTS_STORAGE_AUTH": (
+                '{"NODE_PORTS_STORAGE_LOGIN": "login", '
+                '"NODE_PORTS_STORAGE_PASSWORD": "passwd", '
+                '"NODE_PORTS_STORAGE_HOST": "host", '
+                '"NODE_PORTS_STORAGE_PORT": "42"}'
+            )
+        },
+    ],
+)
+def test_settings_with_node_ports_storage_auth(
+    mock_environment: EnvVarsDict, monkeypatch: pytest.MonkeyPatch, envs: dict[str, str]
+):
+    setenvs_from_dict(monkeypatch, envs)
```

Comment:
TIP: for all pydantic models we have automatic testing with the examples you add in the Config. 
I would rather add this parametrization as example Config of the settings and include in this service . This is the code, adapted from `models_library`


```python
import json
from typing import Any

import sicore_service_dynamic_sidecar
import pytest
from pydantic import BaseModel
from pytest_simcore.pydantic_models import walk_model_examples_in_package


@pytest.mark.parametrize(
    "model_cls, example_name, example_data",
    walk_model_examples_in_package(sicore_service_dynamic_sidecar),
)
def test_all_models_library_models_config_examples(
    model_cls: type[BaseModel], example_name: int, example_data: Any
):
    assert model_cls.parse_obj(
        example_data
    ), f"Failed {example_name} : {json.dumps(example_data)}"
```

Reply:
I have adjusted the test a bit, after your comment.

I am not writing a unit test here.
This is more of an "integration" test, I want to that the storage settings with authentication are read correctly and are different form the storage settings without any authentication.